app/services/event.service.ts

```diff
@@ -0,0 +1,89 @@
+// angular
+import { Injectable, NgZone } from "@angular/core";
+import { BehaviorSubject } from "rxjs/Rx";
+import * as appSettingsModule from "application-settings";
+
+// nativescript
+import * as httpModule from "http";
+
+// app
+import { IEvent } from "../shared/interfaces";
+import { EventModel } from './../shared/event.model';
+import { SessionTypes } from "../shared/static-data";
+import { FavoritesService } from "./favorites.service";
+import * as fakeDataServiceModule from "./fake-data.service";
+
+@Injectable()
+export class EventService {
+
+	public eventsLoaded = false;
+	public ignoreCache = true;
+	public items: BehaviorSubject<Array<EventModel>> = new BehaviorSubject([]);
+	private _useHttpService: boolean = false;
+	private _allEvents: Array<EventModel> = [];
+
+	constructor(
+		private _zone: NgZone,
+	) { 
+		try {
+			let cahcedEvents = <Array<EventModel>>JSON.parse(appSettingsModule.getString('ALLSESSIONS', '[]'));
```

Comment:
This will overwrite the session data. Save this as "EVENTS" instead.

Reply:
Also, where are you actually saving the data in appSettingsModule? I dont see that code in here.

Reply:
not really utilizing the cache at all for Events, will probably just get rid of it

Reply:
Wouldn't we need to cache the event data?

Reply:
You're right, I should probably fix up the cache and set it up for events like it is setup for sessions.